File: android/app/src/main/java/com/fabricio/challenge/model/Product.java
package com.fabricio.challenge.model;

import android.text.Html;
import android.text.Spanned;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Product {

    @Expose
    @SerializedName("id")
    private Integer id;

    @Expose
    @SerializedName("categoria")
    private Category category;

    @Expose
    @SerializedName("descricao")
    private String description;

    @Expose
    @SerializedName("nome")
    private String name;

    @Expose
    @SerializedName("precoDe")
    private Double priceOriginal;

    @Expose
    @SerializedName("precoPor")
    private Double priceDiscount;

    @Expose
    @SerializedName("urlImagem")
    private String urlImage;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Category getCategory() {
        return category;
    }

    public void setCategory(Category category) {
        this.category = category;
    }

    public String getDescription() {
        return description;
    }

    public Spanned getDescriptionFormatted() {
        return Html.fromHtml(description);
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Double getPriceOriginal() {
        return priceOriginal;
    }

    public void setPriceOriginal(Double priceOriginal) {
        this.priceOriginal = priceOriginal;
    }

    public Double getPriceDiscount() {
        return priceDiscount;
    }

    public void setPriceDiscount(Double priceDiscount) {
        this.priceDiscount = priceDiscount;
    }

    public String getUrlImage() {
        return urlImage;
    }

    public void setUrlImage(String urlImage) {
        this.urlImage = urlImage;
    }
}

File: android/Lodjinha/app/src/main/java/com/example/lodjinha/data/classes/Product.kt
package com.example.alodjinha.data.classes

import com.example.alodjinha.data.classes.Category
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Product {

    @SerializedName("id")
    @Expose
    private var id: Int? = null

    @SerializedName("nome")
    @Expose
    private var name: String? = null

    @SerializedName("urlImagem")
    @Expose
    private var urlImage: String? = null

    @SerializedName("descricao")
    @Expose
    private var description: String? = null

    @SerializedName("precoDe")
    @Expose
    private var price: Double? = null

    @SerializedName("precoPor")
    @Expose
    private var priceFor: Double? = null

    @SerializedName("categoria")
    @Expose
    private var category: Category? = null


    fun setId(Id: Int){ id = Id }

    fun getId() = id

    fun setName(Name: String){ name = Name }

    fun getName() = name

    fun setUrlImage(UrlImage: String){ urlImage = UrlImage}

    fun getUrlImage() = urlImage

    fun setDescription(Description: String){ description = Description }

    fun getDescription() = description

    fun setPrice(Price: Double){ price = Price }

    fun getPrice() = price

    fun setPriceFor(PriceFor: Double){ priceFor = PriceFor }

    fun getPriceFor() = priceFor

    fun setCategoryId(CatId: Int){
        category?.setId(CatId)
    }
    fun getCategoryId() = category?.getId()

    fun setCategoryDescription(CatDesc: String){
        category?.setDescription(CatDesc)
    }
    fun getCategoryDescription() = category?.getDescription()

    fun setCategoryUrlImage(CatUrl: String){
        category?.setUrlImage(CatUrl)
    }
    fun getCategoryUrlImage() = category?.getUrlImage()



}